www/js/diary/LeafletView.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useRef, useState } from "react";
+import { angularize } from "../angular-react-helper";
+import { object, string } from "prop-types";
+import { View } from "react-native";
+
+const LeafletView = ({ geojson, opts }) => {
+
+  const mapRef = useRef(null);
+
+  useEffect(() => {
+    const map = L.map(mapRef.current, opts || {});
+    L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+    }).addTo(map);
+    const gj = L.geoJson(geojson.data, {
+      pointToLayer: pointToLayer,
+      style: (feature) => feature.style
+    }).addTo(map);
+    const gjBounds = gj.getBounds().pad(0.2);
+    map.fitBounds(gjBounds);
+  }, []);
+
+  return (
+    <View>
+      <div id="map" ref={mapRef} style={{width: '100%', height: '100%'}}></div>
+    </View>
+  );
+};
+
+const pointIcon = L.divIcon({className: 'leaflet-div-icon', iconSize: [0, 0]});
+const startIcon = L.divIcon({className: 'leaflet-div-icon-start', iconSize: [18, 18], html: '<div class="leaflet-div-ionicon leaflet-div-ionicon-start">★</div>'});
+const stopIcon = L.divIcon({className: 'leaflet-div-icon-stop', iconSize: [18, 18], html: '<div class="leaflet-div-ionicon leaflet-div-ionicon-stop">⚑</div>'});
+const pointToLayer = (feature, latlng) => {
+  switch(feature.properties.feature_type) {
+    case "start_place": return L.marker(latlng, {icon: startIcon});
+    case "end_place": return L.marker(latlng, {icon: stopIcon});
+    case "stop": return L.circleMarker(latlng);
+    case "location": return L.marker(latlng, {icon: pointIcon});
```

Comment:
Honestly, I am not really sure and could use guidance. This code was in services.js before. https://github.com/e-mission/e-mission-phone/blob/ee4904dea02abd703689918927af53f702c2569b/www/js/diary/services.js#L344

The only markers I know of that are actually being shown are the start and end place.

I was sure about removing 'incident', but I left 'stop' and 'location' there in case we ever want to display intermediate markers during the trip.

Reply:
I don't know either. This goes back to the very first commit from 2016 that turned on the diary screen in the hybrid app for the first time. And I don't have any explanations. I think it should be safe to remove given that we show all locations in a linestring and we have a separate marker for incidents
https://github.com/e-mission/e-mission-phone/commit/df2c8520140812c51c2178bc56e65a2d0e37cbc8

Reply:
addressed in #980 - removed `"location"`. kept `"stop"`, for maybe later use, but commented out since it's not being used now.